### PR TITLE
Nightshade 5.3.2 upgrade (ome-dundeeomero & ns-web)

### PR DESCRIPTION
**Pull request by kennethgillen**
tested in infra-testpr - produces an upgraded OMERO.web 5.3.2 server.

### CODE REVIEW

**Comment by kennethgillen** [comment]
All that was required to upgrade OMERO version to both decoupled webserver and OMERO server:

  * tell the role openmicroscopy.omero-server we want to upgrade
  * increment the version numbers

Then run the playbook!:

# Wed 17 May 2017 18:01:42 BST:  ansible-playbook -l ns-web.openmicroscopy.org -i /Users/kenny/forks/management_tools/ansible/inventory/prod-hosts playbook.yml --ask-vault-pass --ask-become-pass

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [ns-web.openmicroscopy.org]

TASK [OMERO.web apps | (pre_task) configure systemd for pythonpath - create config folder] ***
ok: [ns-web.openmicroscopy.org]

TASK [OMERO.web apps | (pre_task) configure systemd for pythonpath] ************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.lvm-partition : storage | create logical volume] **********
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.lvm-partition : storage | format] *************************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.lvm-partition : storage | mount] **************************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-common : omero-common | create parent directory] ****
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omego : omego | install python virtualenv] ****************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omego : omego | setup virtualenv] *************************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.basedeps : system packages | install epel repo] ***********
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.basedeps : system packages | basic system utils] **********
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.ice : Load ice variables] *********************************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.ice : zeroc ice | setup repository] ***********************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.ice : zeroc ice | install runtime packages] ***************
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.ice : zeroc ice | install devel packages] *****************
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.ice : zeroc ice | install ice pip dependencies] ***********
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.ice : zeroc ice | pip install packages (ansible < 2.2)] ***
skipping: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.ice : zeroc ice | pip install packages (ansible >= 2.2)] **
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.nginx : system packages | setup upstream nginx stable repo] ***
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.nginx : system packages | setup upstream nginx mainline repo] ***
skipping: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.nginx : system packages | install nginx] ******************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.nginx : nginx | remove default config] ********************
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.nginx : logrotate | configure main file] ******************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-python-deps : omero python deps | distribution packages] ***
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.selinux-utils : selinux | check exists] *******************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.selinux-utils : selinux | check enabled] ******************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.selinux-utils : system packages | set selinux variable] ***
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.selinux-utils : system packages | install selinux utilities] ***
skipping: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.omero-web : omero web | create system user] ***************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | create web directories] ***********
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.omero-web : omero web | install python redis package] *****
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | setup virtualenv] *****************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | selinux booleans] *****************
skipping: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))
skipping: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [openmicroscopy.omero-web : omero web | selinux ports] ********************
skipping: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | is web symlink present] ***********
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | get web version] ******************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | check omero version could be obtained] ***
ok: [ns-web.openmicroscopy.org] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}

TASK [openmicroscopy.omero-web : omero web | checkupgrade] *********************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : debug] ****************************************
ok: [ns-web.openmicroscopy.org] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}

TASK [openmicroscopy.omero-web : omero web | stop omero web] *******************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | configuration script] *************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | configuration 00-omero-web.omero] *
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | install omero web] ****************
changed: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | check web timestamp] **************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | check nginx timestamp] ************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | check if nginx upgrade needed] ****
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : debug] ****************************************
ok: [ns-web.openmicroscopy.org] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}

TASK [openmicroscopy.omero-web : omero web | generate nginx config] ************
changed: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | nginx remove system defaults] *****
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | enable nginx] *********************
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | systemd service] ******************
ok: [ns-web.openmicroscopy.org]

RUNNING HANDLER [openmicroscopy.omero-common : restart nginx] ******************
changed: [ns-web.openmicroscopy.org]

RUNNING HANDLER [openmicroscopy.omero-web : omero-web rewrite omero-web configuration] ***
changed: [ns-web.openmicroscopy.org]

RUNNING HANDLER [openmicroscopy.omero-web : omero-web restart omero-web] *******
changed: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.omero-web : omero web | enable and start web systemd] *****
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.system-monitor-agent : system packages | install epel repo] ***
ok: [ns-web.openmicroscopy.org]

TASK [openmicroscopy.system-monitor-agent : system packages | check-mk agent] **
ok: [ns-web.openmicroscopy.org] => (item=[u'check-mk-agent-1.2.6p16', u'xinetd'])

TASK [openmicroscopy.system-monitor-agent : xinetd | start and enable] *********
ok: [ns-web.openmicroscopy.org]

TASK [Install open-vm-tools if system is a VMware vm] **************************
ok: [ns-web.openmicroscopy.org]

TASK [NGINX - Performance tuning - worker processes] ***************************
ok: [ns-web.openmicroscopy.org]

TASK [NGINX - Performance tuning - worker connections] *************************
ok: [ns-web.openmicroscopy.org]

TASK [NGINX - SSL File Deployment - prepare directory] *************************
ok: [ns-web.openmicroscopy.org]

TASK [NGINX - SSL File Deployment] *********************************************
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))
ok: [ns-web.openmicroscopy.org] => (item=(censored due to no_log))

TASK [NGINX - SSL Configuration - Additional listen port] **********************
changed: [ns-web.openmicroscopy.org]

TASK [NGINX - SSL Configuration - Rest of SSL section to omero-web.conf] *******
changed: [ns-web.openmicroscopy.org]

TASK [OMERO.web apps | top-level folder] ***************************************
ok: [ns-web.openmicroscopy.org]

TASK [OMERO.web apps | download latest figure] *********************************
skipping: [ns-web.openmicroscopy.org]

TASK [OMERO.web apps | app-name symlinks] **************************************
ok: [ns-web.openmicroscopy.org] => (item={u'dest': u'figure', u'src': u'figure-1.2.1'})

TASK [template] ****************************************************************
ok: [ns-web.openmicroscopy.org]

RUNNING HANDLER [openmicroscopy.omero-common : restart nginx] ******************
changed: [ns-web.openmicroscopy.org]

PLAY RECAP *********************************************************************
ns-web.openmicroscopy.org  : ok=64   changed=8    unreachable=0    failed=0

File: ansible/server-state-playbooks/ome-dundeeomero/playbook.yml [hunk 1]
```diff
@@ -135,7 +135,8 @@
     # Note - had to have these set to `install-mock` to progress role
     # installation before changing config to restored DB from other system.
     - role: openmicroscopy.omero-server
-      omero_server_release: 5.3.1
+      omero_server_release: 5.3.2
+      omero_server_upgrade: True
```
Comment (joshmoore) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: Perhaps just pass this on the command-line?

Comment (kennethgillen) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: Which - the 'upgrade' flag? I'm not sure it does any harm if we're controlling the version to be installed with the `_release` variable. That flag has existed for some time in the ns-web playbook (inserted at the 5.3.0->5.3.1 upgrade: ﻿﻿﻿https://github.com/openmicroscopy/infrastructure/pull/260/files#diff-73638a117d0afc35645cb554e59abdc1R48).

**Comment by kennethgillen** [comment]
Merging as 'current state'.

File: ansible/server-state-playbooks/ome-dundeeomero/playbook.yml (hunk 1 above)
Comment (joshmoore) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: Sorry, I missed this comment before say ":thumbsup:". I may be misremembering but I would consider this flag dangerous on a production system. I suggest reverting it.

Comment (kennethgillen) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: Doesn't the `omero_server_release` pin the version, regardless of the `upgrade` flag?

Comment (joshmoore) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: Yes, I don't think a different version would be installed but I could imagine both your server restarting when you don't want it to or omego attempting other _stuff_.

Comment (kennethgillen) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: Ah, that'd be nice if it quashed all the `restart systemd` calls. Need to get to know the playbook more. That was one of my lingering questions about re-running - i.e. I'm not sure I'd be happy rerunning this outside downtime, and therefore wouldn't straight off want it running on a daily cron.

Comment (joshmoore) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: I'd imagine running with `-C` would be fine, but if that's not green (with the exception of the pip weirdness) then I definitely wouldn't run it against production, which should be our goal with tidying up.

Comment (kennethgillen) [inline ansible/server-state-playbooks/ome-dundeeomero/playbook.yml]: https://github.com/openmicroscopy/infrastructure/pull/273/commits/e500d83bb44a1b102501080cc23bcb91e6deea35

**Pull request merged**